layer.go

```diff
@@ -138,14 +132,7 @@ func (l *Layer) Close() error {
 		_, file, line, _ := runtime.Caller(1)
 		panic(fmt.Sprintf("%s:%d: Layer closed twice", file, line))
 	}
-	if l.noFun == nil {
-		return errors.New("claircore: Close: uninitialized Layer")
```

Comment:
Do we want to allow users to close uninitialized layers?

Reply:
Good question. Previously, we needed to de-reference a pointer to do the close (well, in addition to the receiver) and now we don't. Leaving it as-is might make failure paths easier?

After looking at callsites, I don't think it actually will, so I'll go back to the previous behavior.